include/EASTL/sort.h

```diff
@@ -738,15 +813,39 @@ namespace eastl
 		// This is because the implementation below makes an assumption about the input
 		// data that quick_sort satisfies but arbitrary data may not.
 		// There is a standalone insertion_sort function. 
-		template <typename RandomAccessIterator, typename Compare>
+		template <typename RandomAccessIterator, typename Compare, typename eastl::enable_if<!eastl::is_lvalue_reference<typename eastl::iterator_traits<RandomAccessIterator>::value_type>::value>::type* = nullptr >
```

Comment:
Sorry for the delay in review.  

Don't want to check if the type has a move-constructor instead of if its an lvalue reference?